game/src/contrib/systems/HealthSystem.java

```diff
@@ -46,12 +54,10 @@ public void update() {
                 .forEach(this::removeDeadEntities);
```

Comment:
Den letzten Filter verstehe ich nicht wirklich. In Zeile 43 filtern wir alle toten Entitäten, in Zeile 54 nehmen wir sie aus dem Spiel. Aber was macht der Filter in Zeile 44?

- Wenn die Entität tot ist und keine Sterbe-Animation hat oder diese noch läuft - dann true und entsprechend in 54: nimm mich raus? Klingt irgendwie falsch?
- Wenn die aktuelle Animation nicht die Sterbe-Animation ist, setze sie. -> Das ist kein Filter, sondern ein Nebeneffekt.
- Und last not least: Wenn die Entität tot ist und die aktuelle(!) Animation durch ist, dann true und in 54 weg damit. Das klingt ungefähr sinnig - könnte sich aber nicht um die Sterbe-Animation handeln.

=> hier macht ein Filter irgendwie ganz viel. Das müsste nochmal in verschiedene Vorgänge gesplittet werden (die eigentlichen Filter und das Setzen der Animation). Zusätzlich sollte es eine Variable mit einer Hilfsfunktion o.ä. geben, die die komplexere Abfrage für den Filter macht mit entsprechenden Kommentaren.

Reply:
Edit: Sorry, der Kommentar klebt irgendwie an der falschen Stelle (unter dem angemerkten Filter).